Handle the fact that gets() no longer exists in glibc >= 2.16

The problem has been fixed upstream, but no release containing the fix
has been made so far.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/lib/stdio.in.h
===================================================================
--- a/lib/stdio.in.h
+++ b/lib/stdio.in.h
@@ -693,12 +693,14 @@
 _GL_CXXALIAS_SYS (gets, char *, (char *s));
 #  undef gets
 # endif
+# if defined gets
 _GL_CXXALIASWARN (gets);
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
 #endif
+#endif
 
 
 #if @GNULIB_OBSTACK_PRINTF@ || @GNULIB_OBSTACK_PRINTF_POSIX@
